src/main/java/frc/robot/subsystems/tank/PoseEstimatorThread.java

```diff
@@ -0,0 +1,116 @@
+package frc.robot.subsystems.tank;
+
+import com.kauailabs.navx.frc.AHRS;
+import com.revrobotics.RelativeEncoder;
+
+import edu.wpi.first.math.MatBuilder;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.estimator.DifferentialDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.Watchdog;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+
+public class PoseEstimatorThread {
+
+    private final PoseEstimatorRunnable runnable;
+
+    public PoseEstimatorThread(AHRS ahrs, RelativeEncoder leftEncoder, RelativeEncoder rightEncoder) {
+        runnable = new PoseEstimatorRunnable(ahrs, leftEncoder, rightEncoder);
+
+        Thread thread = new Thread(runnable);
+        thread.setDaemon(true);
+        thread.start();
+    }
+
+    public Pose2d getPosition() {
+        return runnable.getPosition();
+    }
+
+    public void setPosition(Pose2d pose) {
+        runnable.setPosition(pose);
+    }
+
+    public DifferentialDriveWheelSpeeds getWheelSpeeds() {
+        return runnable.getLastWheelSpeeds();
+    }
+
+    class PoseEstimatorRunnable implements Runnable {
+        private final DifferentialDrivePoseEstimator poseEstimator;
+
+        private final AHRS ahrs;
+
+        private final RelativeEncoder leftEncoder;
+        private final RelativeEncoder rightEncoder;
+
+        private DifferentialDriveWheelSpeeds lastWheelSpeeds;
+
+        public PoseEstimatorRunnable(AHRS ahrs, RelativeEncoder leftEncoder, RelativeEncoder rightEncoder) {
+            this.ahrs = ahrs;
+            this.leftEncoder = leftEncoder;
+            this.rightEncoder = rightEncoder;
+
+            // https://docs.wpilib.org/en/stable/docs/software/advanced-controls/state-space/state-space-pose_state-estimators.html
+            poseEstimator = new DifferentialDrivePoseEstimator(new Rotation2d(), new Pose2d(),
+                    // State measurement standard deviations. X, Y, theta.
+                    new MatBuilder<>(Nat.N5(), Nat.N1()).fill(0.02, 0.02, 0.01, 0.02, 0.02),
+                    // Local measurement standard deviations. Left encoder, right encoder, gyro.
+                    new MatBuilder<>(Nat.N3(), Nat.N1()).fill(0.02, 0.02, 0.01),
+                    // Global measurement standard deviations. X, Y, and theta.
+                    new MatBuilder<>(Nat.N3(), Nat.N1()).fill(0.1, 0.1, 0.01));
+        }
+
+        @Override
+        public void run() {
+            while (true) {
```

Comment:
There's no `Thread.sleep()` in this loop.

Reply:
... yes, because we want to be estimating our pose as often as possible

Reply:
Uhhhh. @e3l you probably should sleep once an a while, and you probably don't need to update _that_ often. 10 Hz should be plenty. Do we know the timing here?

Reply:
Haven't extensively profiled the performance of this thread or how accuracy changes with different update speeds.

We do know that it's more I/O blocked than anything else, which means that having no sleep doesn't mean it'll be expensive or slow down other threads, necessarily, right?